Caltech-IPAC/firefly
# DM-7761 Build LSST Catalog search form using the components from IRSA catalog search form

Build LSST catalog search panel including two tabs, 'Search Catalogs' and 'Load Catalog'
- add 'LSST Catalogs' to the firefly menu. 
- create LSST catalog search panel, 
  . reuse 'CatalogConstraintPanel' (used by IRSA catalog search form) component.
  . making improvement on 'CatalogConstraintPanel' including 
       validation of column search constraints. 
       storing and restoring of column selection and constraint string. 
  . incorporating catalog spatial search features made in DM-7995.
  . For 'Load Catalog', redisplay the filename of last uploaded file by button 'Choose File'. 
    (this feature is applied to all cases which use FileUpload and keep the uploaded filename) 
  
Test: 
- start firefly localhost:8080/firefly
- select 'LSST Catalogs' and the search panel is started
- select LSST catalog, either 'Deep Forced Source' or 'Deep Source' to see the metadata table update in the middle of the panel. 
- enter the spatial method search, select the column and enter the column search constraints in the middle metadata table. 
- click 'search' to get the data from the selected catalog. 

Result: 
All search result in this release is temporary and hard coded. It will be further updated once the catalog data is ready.
